tests/subsys/fs/fat_fs_api/src/test_fat_file_reentrant.c

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2022 Husqvarna AB
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "test_fat.h"
+
+#if IS_ENABLED(CONFIG_FS_FATFS_REENTRANT)
+
+#define REENTRANT_TEST_STACK_SIZE 500
+#define SEMAPHORE_OP_SUCCESS 0
+
+void tlock_mutex(void *p1, void *p2, void *p3);
+
+K_THREAD_STACK_DEFINE(tlock_mutex_stack_area, REENTRANT_TEST_STACK_SIZE);
+struct k_thread tlock_mutex_data;
+struct k_sem mutex_unlocked_sem;
+struct k_sem run_non_thread_sem;
+
+static int test_reentrant_access(void)
+{
+	int res;
+
+	TC_PRINT("\nReentrant tests:\n");
+	zassert_false(k_sem_init(&mutex_unlocked_sem, 0, 1), NULL);
+	zassert_false(k_sem_init(&run_non_thread_sem, 0, 1), NULL);
+
+	/* Start mutex locking thread */
+	k_tid_t tid = k_thread_create(&tlock_mutex_data, tlock_mutex_stack_area,
+			K_THREAD_STACK_SIZEOF(tlock_mutex_stack_area),
+			tlock_mutex,
+			NULL, NULL, NULL,
+			K_PRIO_PREEMPT(0), 0, K_NO_WAIT);
+
+	/* Make sure thread was able to lock mutex */
+	k_sem_take(&run_non_thread_sem, K_FOREVER);
+
+	/* File open should wait here, as the fs is locked. Therefore, automatic switch back to thread */
+	TC_PRINT("Open file\n");
+	res = fs_open(&filep, TEST_FILE, FS_O_CREATE | FS_O_RDWR);
+	zassert_false(res, "Err: File could not be opened [%d]\n", res);
```

Comment:
`zasset_false` expects boolean; you probably would like to use `zassert_ok` here.